src/api/Metadata.js

```diff
@@ -154,26 +176,223 @@ class Metadata extends File {
+            const key = this.getClassificationCacheKey(id);
+            cache.set(key, classification);
+        }
+        return instances;
+    }
+
+    /**
+     * Finds template for a given metadata instance.
+     *
+     * @param {string} id - Box file id
+     * @param {Object} instance - metadata instance
+     * @param {Array} templates - metadata templates
+     * @return {Object|undefined} template for metadata instance
+     */
+    async getTemplateForInstance(
+        id: string,
+        instance: MetadataInstance,
+        templates: Array<MetadataEditorTemplate>,
+    ): Promise<?MetadataEditorTemplate> {
+        const instanceId = instance.$id;
+        const templateKey = instance.$template;
+        const scope = instance.$scope;
+        let template = templates.find(t => t.templateKey === templateKey && t.scope === scope);
+
+        // Enterprise scopes are always enterprise_XXXXX
+        if (scope.startsWith(METADATA_SCOPE_ENTERPRISE) && !template) {
+            // If the template does not exist, it can be a template from another
+            // enterprise because the user is viewing a collaborated file.
+            const crossEnterpriseTemplate = await this.getTemplates(id, scope, instanceId);
+            // The API always returns an array of at most one item
+            template = crossEnterpriseTemplate[0]; // eslint-disable-line
+        }
+
+        return template;
+    }
+
+    /**
+     * Creates and returns metadata editors.
+     *
+     * @param {string} id - Box file id
+     * @param {Array} instances - metadata instances
+     * @param {Object} customPropertiesTemplate - custom properties template
+     * @param {Array} enterpriseTemplates - enterprise templates
+     * @param {Array} globalTemplates - global templates
+     * @param {boolean} canEdit - metadata editability
+     * @return {Array} metadata editors
+     */
+    async getEditors(
+        id: string,
+        instances: Array<MetadataInstance>,
+        customPropertiesTemplate: MetadataEditorTemplate,
+        enterpriseTemplates: Array<MetadataEditorTemplate>,
+        globalTemplates: Array<MetadataEditorTemplate>,
+        canEdit: boolean,
+    ): Promise<Array<MetadataEditor>> {
+        // All usable templates for metadata instances
+        const templates: Array<MetadataEditorTemplate> = [customPropertiesTemplate].concat(
+            enterpriseTemplates,
+            globalTemplates,
+        );
+
+        // Filter out skills and classification
+        // let filteredInstances = this.extractSkills(id, instances);
+        const filteredInstances = this.extractClassification(id, instances);
+
+        // Create editors from each instance
+        const editors: Array<MetadataEditor> = [];
+        await Promise.all(
+            filteredInstances.map(async instance => {
+                const template: ?MetadataEditorTemplate = await this.getTemplateForInstance(id, instance, templates);
```

Comment:
if one of the underlying `getTemplates` API call fails for whatever reason and throws an error, what is the expected behavior here?

Reply:
There is a catch in there that can re-throw, which will be caught here and the errorHandler() will be called. Notice this is surrounded with try catch